Snowflake.Data/Core/Session/SessionPool.cs

```diff
@@ -143,6 +153,8 @@ private SFSession NewSession(String connectionString, SecureString password)
             try
             {
                 var session = s_sessionFactory.NewSession(connectionString, password);
+                if (_pooling)
+                    _busySessionsCounter.Increase();
```

Comment:
I would increase the counter after we open a session (for the sake of any error occurred)

Reply:
sessionPool offers in their API:
- GetSession() - here we increment busySessions count
- AddSession() - which is rather "return session to the pool as an idle session" which is decreasing busySessions count
There is no place for open() in this api. Remember that for both: idle and busy sessions, the session is already opened.